services/src/main/java/org/exoplatform/officeonline/rest/WOPIService.java

```diff
@@ -38,32 +38,74 @@
 
-  /** The Constant LOG. */
-  protected static final Log          LOG          = ExoLogger.getLogger(WOPIService.class);
+  protected enum Operation {
+    GET_LOCK, GET_SHARE_URL, LOCK, PUT, PUT_RELATIVE, REFRESH_LOCK, RENAME_FILE, UNLOCK
+  }
+
+  /** The Constant FILE_CONVERSION. */
+  protected static final String                         FILE_CONVERSION   = "X-WOPI-FileConversion";
+
+  /** The Constant ITEM_VERSION. */
+  protected static final String                         ITEM_VERSION      = "X-WOPI-ItemVersion";
+
+  /** The Constant LOCK. */
+  protected static final String                         LOCK              = "X-WOPI-Lock";
+
+  /** The Constant MAX_EXPECTED_SIZE. */
+  protected static final String                         MAX_EXPECTED_SIZE = "X-WOPI-MaxExpectedSize";
+
+  /** The Constant OLD_LOCK. */
+  protected static final String                         OLD_LOCK          = "X-WOPI-OldLock";
+
+  /** The Constant OVERRIDE. */
+  protected static final String                         OVERRIDE          = "X-WOPI-Override";
+
+  /** The Constant PROOF. */
+  protected static final String                         PROOF             = "X-WOPI-Proof";
+
+  /** The Constant PROOF_OLD. */
+  protected static final String                         PROOF_OLD         = "X-WOPI-ProofOld";
+
+  /** The Constant RELATIVE_TARGET. */
+  protected static final String                         RELATIVE_TARGET   = "X-WOPI-RelativeTarget";
+
+  /** The Constant REQUESTED_NAME. */
+  protected static final String                         REQUESTED_NAME    = "X-WOPI-RequestedName";
+
+  /** The Constant SUGGESTED_TARGET. */
+  protected static final String                         SUGGESTED_TARGET  = "X-WOPI-SuggestedTarget";
+
+  /** The Constant URL_TYPE. */
+  protected static final String                         URL_TYPE          = "X-WOPI-UrlType";
+
+  /** The Constant TIMESTAMP. */
+  protected static final String                         TIMESTAMP         = "X-WOPI-TimeStamp";
 
   /** The Constant ACCESS_TOKEN. */
-  protected static final String       ACCESS_TOKEN = "access_token";
+  protected static final String                         ACCESS_TOKEN      = "access_token";
 
   /** The Constant API_VERSION. */
-  protected static final String       API_VERSION  = "1.1";
+  protected static final String                         API_VERSION       = "1.1";
+
+  /** The Constant LOG. */
+  protected static final Log                            LOG               = ExoLogger.getLogger(WOPIService.class);
 
   /** The editor service. */
-  protected OfficeOnlineEditorService editorService;
+  protected final OfficeOnlineEditorService.WOPIService wopiSPI;
 
   /**
    * Instantiates a new WOPI service.
    *
    * @param editorService the editor service
    */
   public WOPIService(OfficeOnlineEditorService editorService) {
-    this.editorService = editorService;
+    this.wopiSPI = editorService.getWOPIService();
```

Comment:
This class name is WOPIService, a method editorService.getWOPIService() sounds a but misleading, let's change it to getWOPIConnector() and related internal class name change to WOPIConnector also.